### PR TITLE
contributing_to_docs/term_glossary.adoc fixups

### PR DESCRIPTION
**Pull request by Miciah**
Rewrite the introductory paragraph to be clearer (guidelines for what?) and more comprehensive (components and objects).

Sort out a mix-up of the definitions of action and authorization.

Try to make the definition of "application" a little clearer, delete a confusing example, and fix an example that was a sentence fragment.

Define "apiserver", "cluster", "deployment controller", "scheduler", and "SkyDNS".

Fix usage for "identity".

Note that "OpenShift master" is generally preferred over "Kubernetes master".

Resolve some inconsistency in usage texts.

Fix some minor punctuation issues and unclear writing throughout.

### CODE REVIEW

File: contributing_to_docs/term_glossary.adoc [hunk 1]
```diff
@@ -84,6 +89,13 @@ Usage: build(s) as appropriate
...
 ''''
 
+=== cluster
+
+Usage: cluster
+
+The collection of controllers, pods, and services and related DNS and networking
+routing configuration that are defined on the system.
+''''
 === container
```
Comment (adellape) [inline contributing_to_docs/term_glossary.adoc]: Rendering here is wacky. Need an empty line above the heading I think.

Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: There are a lot of places where we have a '''' line adjacent to an === line.  Shall I fix all such instances?

Comment (Miciah) [inline contributing_to_docs/term_glossary.adoc]: Oh, I probably just need to add a blank line above the ''''.

File: contributing_to_docs/term_glossary.adoc [hunk 2]
```diff
@@ -227,12 +252,27 @@ communities.
...
+expected to be running are actually running.
+''''
+=== SkyDNS
+
+Usage: SkyDNS
+
+Component of the Kubernetes master or OpenShift master that provides
+cluster-wide DNS resolution of internal hostnames for services and pods.
+''''
 === Source-to-Image (S2I)
```
Comment (adellape) [inline contributing_to_docs/term_glossary.adoc]: Same here wrt blank line.

File: contributing_to_docs/term_glossary.adoc [hunk 3]
```diff
@@ -227,12 +253,27 @@ communities.
...
 ''''
+=== scheduler
+
+Usage: scheduler(s) as appropriate
+
+Component of the Kubernetes master or OpenShift master that manages the state of
+the system, places pods on nodes, and ensures that all containers that are
+expected to be running are actually running.
+''''
+=== SkyDNS
```
Comment (adellape) [inline contributing_to_docs/term_glossary.adoc]: Oops here too.

**Comment by adellape** [comment]
LGTM after latest push. Waiting on Travis...............

**Pull request merged**